zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -95,6 +96,96 @@ public InterpreterFactory(ZeppelinConfiguration conf, InterpreterOption defaultO
     init();
   }
 
+  public boolean loadDynamicInterpreter(String intpGroupName, String intpName, String artifact,
+                                        String intpClassName) {
+    return loadDynamicInterpreter(intpGroupName, intpName, artifact, intpClassName, null, false);
+  }
+
+  public boolean loadDynamicInterpreter(String intpGroupName, String intpName, String artifact,
+                                        String intpClassName, String repositoryUrl,
+                                        boolean isSnapShotRepo) {
+    String[] artifactItem = artifact.split(":");
+    String interpreterDesPath = conf.getString(ConfVars.ZEPPELIN_INTERPRETER_REPO_DIR);
+    interpreterDesPath += "/" + intpGroupName + "/" + intpName + "/";
+
+    String interpreterLoadPath = conf.getRelativeDir(ConfVars.ZEPPELIN_INTERPRETER_REPO_DIR);
+    interpreterLoadPath += "/" + intpGroupName + "/" + intpName;
+
+    if (artifactItem.length <= 0) {
+      logger.error("invalid artifact [" + artifact + "]");
+      return false;
+    }
+
+    try {
+      if (repositoryUrl != null) {
+        depResolver.addRepo("dyInterpreterRepo", repositoryUrl, isSnapShotRepo);
+      }
+      logger.info("interpreter- path {}", interpreterLoadPath);
+      depResolver.load(artifact, interpreterDesPath);
+      setDynamicInterpreter(intpClassName, interpreterLoadPath);
+    } catch (Exception e) {
+      logger.error(e.getMessage());
+      return false;
+    }
+    return true;
+  }
+
+  public boolean unloadDynamicInterpreter(String intpGorupName, String intpName) {
+    try {
+      remove(intpName);
+    } catch (Exception e) {
+      logger.error(e.getMessage());
```

Comment:
use `logger.error("BOOM!", e);` instead?
http://stackoverflow.com/questions/15805460/how-to-print-an-exception-using-logger